### PR TITLE
improve probes lookup by dropping use of regex

### PR DESCRIPTION
**Pull request by mmarchini**
This PR alone won't solve the performance issues, but combined with #410 I was able to reduce startup time when looking up a large library from almost 20s to about 300ms.

Also, let's keep #526 open since there are other performance improvements we can do (for example, only expand wildcards if there's a wildcard in our probe name).

### CODE REVIEW

**Comment by mmarchini** [comment]
Not fully working yet. Will take a look into this again tomorrow.

**Comment by mmarchini** [comment]
I should probably point out that we won't be able to use regex in probe definitions with this PR, but I don't think that's an issue as long as we have wildcards working.

**Comment by mmarchini** [comment]
Ok, should be good to merge if there's no objections to https://github.com/iovisor/bpftrace/pull/557#issuecomment-485921116 and if the CI is green.

File: src/bpftrace.cpp [hunk 1]
```diff
@@ -153,25 +153,30 @@ std::set<std::string> BPFtrace::find_wildcard_matches(const std::string &prefix,
...
+    if (!full_prefix.empty()) {
+      if (line.find(full_prefix, 0) != 0)
+        continue;
+      line = line.substr(full_prefix.length());
     }
+
+    if (!wildcard_match(line, tokens, start_wildcard, end_wildcard))
+      continue;
+
+    if (line.find(".part.") != std::string::npos)
```
Comment (ajor) [inline src/bpftrace.cpp]: Could you keep the comment on this line please? I didn't understand what it was for until I looked at the old code

**Comment by dalehamel** [comment]
Not sure if we should fix this here too, but list and attach should use the same method to match and filter probes, but it is currently duplicated (and inconsistent):

https://github.com/iovisor/bpftrace/blob/master/src/list.cpp#L116-L139
https://github.com/iovisor/bpftrace/blob/master/src/bpftrace.cpp#L152

Probably not in the scope of this PR, but we should cut an issue to fix that too.

**Comment by mmarchini** [comment]
@dalehamel yes, we need to improve consistency between attaching and listing probes, but it should be worked on another PR.

File: src/bpftrace.cpp (hunk 1 above)
Comment (mmarchini) [reply src/bpftrace.cpp]: Done

**Comment by mmarchini** [comment]
This is good on my side.

**Comment by dalehamel** [comment]
> yes, we need to improve consistency between attaching and listing probes, but it should be worked on another PR.
 
yup, I cut #565 to address that

**Comment by brendangregg** [comment]
Thanks!

**Comment by ajor** [comment]
Nice tests!

**Pull request merged**